samtranslator/model/api/api_generator.py

```diff
@@ -204,20 +206,81 @@ def _construct_stage(self, deployment, swagger):
         stage.TracingEnabled = self.tracing_enabled
 
         if swagger is not None:
-            deployment.make_auto_deployable(stage, self.remove_extra_stage, swagger)
+            deployment.make_auto_deployable(stage, self.remove_extra_stage, swagger, self.domain)
 
         if self.tags is not None:
             stage.Tags = get_tag_list(self.tags)
 
         return stage
 
+    def _construct_api_domain(self, rest_api):
+        """
+        Constructs and returns the ApiGateway Domain and BasepathMapping
+        """
+        if self.domain is None or \
+           self.domain.get('DomainName') is None or \
+           self.domain.get('CertificateArn') is None:
+            return None, None
+
+        logical_id = logical_id_generator.LogicalIdGenerator("", self.domain).gen()
+
+        domain = ApiGatewayDomainName('ApiGatewayDomainName' + logical_id,
+                                      attributes=self.passthrough_resource_attributes)
+        domain.DomainName = self.domain.get('DomainName')
+        endpoint = self.domain.get('EndpointConfiguration')
+
+        if endpoint is None or endpoint not in ['EDGE', 'REGIONAL']:
+            endpoint = 'REGIONAL'
+
+        if endpoint == 'REGIONAL':
+            domain.RegionalCertificateArn = self.domain.get('CertificateArn')
+        else:
+            domain.CertificateArn = self.domain.get('CertificateArn')
+
+        domain.EndpointConfiguration = {"Types": [endpoint]}
+
+        # Create BasepathMappings
+        if self.domain.get('BasePath') and isinstance(self.domain.get('BasePath'), string_types):
+            basepaths = [self.domain.get('BasePath')]
+        elif self.domain.get('BasePath') and isinstance(self.domain.get('BasePath'), list):
+            basepaths = self.domain.get('BasePath')
+            # Cloudformation restricts you from adding '/' with other paths. '/' is also the default basepath value
+            if '/' in basepaths:
+                basepaths = None
+        else:
+            basepaths = None
```

Comment:
Here, it looks like `basepaths = None` in both if and else section. instead of if-else we can just have `basepaths = None`. I am not sure if i am understanding this correct